modules/lightmapper_rd/lm_compute.glsl

```diff
@@ -352,27 +357,70 @@ void main() {
 			continue; //no need to do anything
 		}
 
-		if (trace_ray(position + light_dir * params.bias, light_pos) == RAY_MISS) {
-			vec3 light = lights.data[i].color * lights.data[i].energy * attenuation;
-			if (lights.data[i].static_bake) {
-				static_light += light;
-#ifdef USE_SH_LIGHTMAPS
+		float penumbra = 0.0;
+		if (lights.data[i].size > 0.0) {
+			vec3 light_to_point = -light_dir;
+			vec3 aux = light_to_point.y < 0.777 ? vec3(0.0, 1.0, 0.0) : vec3(1.0, 0.0, 0.0);
+			vec3 light_to_point_tan = normalize(cross(light_to_point, aux));
+			vec3 light_to_point_bitan = normalize(cross(light_to_point, light_to_point_tan));
+
+			const uint shadowing_rays_check_penumbra_denom = 2;
+			uint shadowing_ray_count = params.ray_count;
+
+			uint hits = 0;
+			uint noise = random_seed(ivec3(atlas_pos, 43573547 /* some prime */));
+			vec3 light_disk_to_point = light_to_point;
+			for (uint j = 0; j < shadowing_ray_count; j++) {
+				// Optimization:
+				// Once already traced an important proportion of rays, if all are hits or misses,
+				// assume we're not in the penumbra so we can infer the rest would have the same result
+				if (j == shadowing_ray_count / shadowing_rays_check_penumbra_denom) {
+					if (hits == j) {
+						// Assume totally lit
+						hits = shadowing_ray_count;
+						break;
+					} else if (hits == 0) {
+						// Assume totally dark
+						hits = 0;
+						break;
+					}
+				}
 
-				float c[4] = float[](
-						0.282095, //l0
-						0.488603 * light_dir.y, //l1n1
-						0.488603 * light_dir.z, //l1n0
-						0.488603 * light_dir.x //l1p1
-				);
+				float r = randomize(noise);
+				float a = randomize(noise) * 2.0 * PI;
+				vec2 disk_sample = (r * vec2(cos(a), sin(a))) * soft_shadowing_disk_size;
```

Comment:
Should the disk be uniformly sampled? If so, then ``r`` should be ``sqrt(randomize(noise))``

Reply:
Probably yes, @JFonS what do you think?

Reply:
@techiepriyansh Not necessarily. The technique in the article is an approximation for real-time soft shadows (using shadow-maps) but in our case we are sampling a spherical light.

The truth is, we are also using an approximation of the [proper spherical light sampling](https://schuttejoe.github.io/post/arealightsampling/), but it works just fine inmost cases. There might be some differences in very large lights, but nothing really noticeable.

To me intuitively, it makes sense to have more samples toward the center of the disk, as its the projection of a sphere but I'd have to do the math and double-check. In any case, I would test the two proposed distributions and see if it makes a difference. I'm pretty sure the difference will be barely noticeable for reasonable light sizes.

Reply:
For future reference, we are going with the non-sqrt version. This will sample the disk more heavily towards the centre.